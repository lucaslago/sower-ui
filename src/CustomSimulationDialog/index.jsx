import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import ValidatedTextField from '../components/ValidatedTextField';

export default class CustomSimulationDialog extends Component {
  constructor(props) {
    super(props);
    this.state = {
      intervalValue: '',
      intervalDisabled: false,
      intervalError: false,
      descriptionValue: '',
      descriptionDisabled: false,
      descriptionError: false,
      positionsValue: '',
      positionsDisabled: false,
      positionsErrorText: '',
      submitDisabled: true,
    };

    this.setInterval = this.setInterval.bind(this);
    this.setDescription = this.setDescription.bind(this);
    this.setPositions = this.setPositions.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  setInterval(event) {
    const value = event.target.value;
    this.setState({
      intervalValue: value,
      intervalError: value ? false : true,
      submitDisabled: !(value && this.state.positionsValue),
    });
  }

  setDescription(event) {
    this.setState({ descriptionValue: event.target.value });
  }

  setPositions(event) {
    const value = event.target.value;
    this.setState({
      positionsValue: value,
      positionsErrorText: value ? '' : 'Positions is required',
      submitDisabled: !(value  && this.state.intervalValue),
    });
  }

  isValidForm() {
    return !this.state.positionsValue || !this.state.intervalValue;
  }

  handleSubmit() {
    if (this.isValidForm) {
      return this.setState({
        intervalError: true,
        positionsErrorText: 'Positions is required',
      });
    }
  }

  render() {
    const actions = [
      <FlatButton
        label="Cancel"
        primary={true}
        onTouchTap={this.props.handleClose}
      />,
      <FlatButton
        label="Submit"
        primary={true}
        disabled={this.state.submitDisabled}
        onTouchTap={this.handleSubmit}
      />,
    ];

    return(
      <div>
        <Dialog
          title="Create Custom Simulation"
          actions={actions}
          modal={true}
          open={this.props.open}
        >
          <ValidatedTextField
            label="Interval (milliseconds)"
            type="number"
            fullWidth={false}
            fieldValue={this.state.intervalValue}
            disabled={this.state.intervalDisabled}
            errorText="Interval is required"
            showValidationError={this.state.intervalError}
            handleChange={this.setInterval}
          /><br/>
          <ValidatedTextField
            label="Description"
            type="text"
            fullWidth={false}
            fieldValue={this.state.descriptionValue}
            disabled={this.state.descriptionDisabled}
            showValidationError={this.state.descriptionError}
            handleChange={this.setDescription}
          /><br/>
          <TextField
            hintText="Array of positions [{...}, {...}, ...]"
            multiLine={true}
            rows={2}
            rowsMax={5}
            fullWidth={true}
            disabled={this.state.positionsDisabled}
            value={this.state.positionsValue}
            errorText={this.state.positionsErrorText}
            type="hidden"
            onChange={this.setPositions}
          />
        </Dialog>
      </div>
    );
  }
}
