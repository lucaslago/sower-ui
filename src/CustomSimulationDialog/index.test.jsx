import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import CustomSimulationDialog from './index';

describe('<CustomSimulationDialog />', () => {
  let wrapper;

  const authService = {
    getToken: jest.fn().mockReturnValue('Basic 12314BAb'),
  };

  const simulationService = {
    start: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
    stop: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
    create: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
  };

  const handleClose = jest.fn();

  beforeEach(() => {
    wrapper = mount(
      <MuiThemeProvider>
        <CustomSimulationDialog
          open
          trackerId="123340901BAJD"
          handleClose={handleClose}
          authService={authService}
          simulationService={simulationService}
        />
      </MuiThemeProvider>,
    );
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
