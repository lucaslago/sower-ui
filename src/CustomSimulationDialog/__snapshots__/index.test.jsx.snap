exports[`<CustomSimulationDialog /> renders correctly 1`] = `
<MuiThemeProvider>
  <CustomSimulationDialog
    authToken="Basic 12313ABDKS"
    handleClose={[Function]}
    open={true}
    simulationService={
      Object {
        "create": [Function],
        "start": [Function],
        "stop": [Function],
      }
    }
    trackerId="123340901BAJD">
    <div>
      <Dialog
        actions={
          Array [
            <FlatButton
              disabled={false}
              label="Cancel"
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              primary={true}
              secondary={false} />,
            <FlatButton
              disabled={true}
              label="Submit"
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              primary={true}
              secondary={false} />,
          ]
        }
        actionsContainerStyle={
          Object {
            "textAlign": "right",
          }
        }
        autoDetectWindowHeight={true}
        autoScrollBodyContent={true}
        modal={true}
        open={true}
        repositionOnUpdate={true}
        title="Create Custom Simulation">
        <RenderToLayer
          open={true}
          render={[Function]}
          useLayerForClickAway={false} />
      </Dialog>
      <Snackbar
        autoHideDuration={4000}
        message=""
        onRequestClose={[Function]}
        open={false}
        style={
          Object {
            "textAlign": "center",
          }
        }>
        <ClickAwayListener
          onClickAway={null}>
          <div
            open={false}
            style={
              Object {
                "bottom": 0,
                "display": "flex",
                "left": "50%",
                "muiPrepared": true,
                "position": "fixed",
                "textAlign": "center",
                "transform": "translate(-50%, 48px)",
                "transition": "transform 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, visibility 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "visibility": "hidden",
                "zIndex": 2900,
              }
            }>
            <WithWidth
              message=""
              open={false}>
              <EventListener
                onResize={[Function]}
                target="window">
                <SnackbarBody
                  message=""
                  open={false}
                  width={3}>
                  <div
                    style={
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.87)",
                        "borderRadius": 2,
                        "flexGrow": 0,
                        "fontFamily": "Roboto, sans-serif",
                        "height": 48,
                        "lineHeight": "48px",
                        "maxWidth": 568,
                        "minWidth": 288,
                        "muiPrepared": true,
                        "padding": "0 24px",
                        "width": "auto",
                      }
                    }
                    width={3}>
                    <div
                      style={
                        Object {
                          "color": "#ffffff",
                          "fontSize": 14,
                          "muiPrepared": true,
                          "opacity": 0,
                          "transition": "opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }>
                      <span />
                    </div>
                  </div>
                </SnackbarBody>
              </EventListener>
            </WithWidth>
          </div>
        </ClickAwayListener>
      </Snackbar>
    </div>
  </CustomSimulationDialog>
</MuiThemeProvider>
`;
